README.md

```diff
@@ -69,6 +69,14 @@ through a [`StakingRewards`](#stakingrewards) contract.
 - Update `setRewardsDuration()` to support changing the reward duration during an active distribution.
   - This allows the duration to be changed even when new reward is added immediately after the end of a distribution
     period, which would be the case when `rewardDistribution` is a smart contract distributing new reward every period.
+- Prevent `rewardsToken` and `stakingToken` from being the same token.
+  - Instead of using the _pull pattern_, this contract expect the address with the `rewardsDistribution` role to
+    transfer the tokens to it and call `notifyRewardAmount` with the correct value.
+  - There is a possible scenario where the amount of tokens transferred to the contract is less than the amount used as
+    a parameter in `notifyRewardAmount`. In such case, if the staking token and rewards token are the same, the staked
+    tokens would be incorporated into the new reward period, causing stakers to lose their capital.
+
+Full diff is referenced [here](./src/synthetix/StakingRewards.sol#L8).
```

Comment:
This link does not work.

Honestly I'd just remove the diff links from everywhere and would just leave the permalink to the original code (in one place). Anyone can do the diff given the permalink. 
IMO the code should be easily maintainable, which means not having to look for many places to update the documentation on each change.

Reply:
Done.

Reply:
I meant removing also from StakingRewards.sol.